### PR TITLE
Types page improvements/DOC-276

### PR DESCRIPTION
**Pull request by kailash-manasarovar**
- Improvements to the JS API docs Types page, renamed Types and interfaces.

### CODE REVIEW

File: docs/api/js/types.md [hunk 1]
```diff
@@ -1,226 +1,303 @@
...
-    ethExecuteTxHash?: string;
-}
-```
+# Types and interfaces
+
+## `AccountAbstractionVersion`
+
+Enumerated list of account abstraction versions.
+
+- None = `0`
```
Comment (StanislavBreadless) [inline docs/api/js/types.md:7]: Let's add a bit more explanation

```suggestion
- None = `0` (used for contracts that are not accounts)
```

File: docs/api/js/types.md [hunk 2]
```diff
@@ -1,226 +1,303 @@
...
+- `fee`: `BigNumberish`;
+- `initiatorAddress`: `Address`;
+- `receivedAt`: `Date`;
+- `ethCommitTxHash?`: `string`;
+- `ethProveTxHash?`: `string`;
+- `ethExecuteTxHash?`: `string`;
+
+## `TransactionReceipt`
+
+Interface representation of transaction receipt containing various fields.
```
Comment (StanislavBreadless) [inline docs/api/js/types.md:271]: maybe mention that extends the ethers.js `TransactionReceipt` (the fields listed below are just the ones that were added to the `TransactionReceipt` from ethers.js)

File: docs/api/js/types.md [hunk 3]
```diff
@@ -1,226 +1,303 @@
...
+
+- `l1BatchNumber`: `number`;
+- `l1BatchTxIndex`: `number`;
+- `logs`: `Array<Log>`;
+- `l2ToL1Logs`: `Array<L2ToL1Log>`;
+
+## `TransactionRequest`
+
+Transaction request type extended from the Ethers `providers.TransactionRequest which adds an optional field for EIP-712 transactions.
```
Comment (StanislavBreadless) [inline docs/api/js/types.md:280]:
```suggestion
Transaction request type extended from the Ethers `providers.TransactionRequest` which adds an optional field for EIP-712 transactions.
```

**Review by StanislavBreadless — COMMENTED** [review]
Looks good, a few suggestions

File: docs/api/js/types.md (hunk 2 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md]: Added @StanislavBreadless

**Review by StanislavBreadless — APPROVED** [review]

File: docs/api/js/types.md [hunk 4]
```diff
@@ -1,226 +1,303 @@
...
+- Full = `0`
+- Rollup = `1`
+
+## `PriorityQueueType`
+
+Enumerated list of priority queue types.
+
+- Deque = `0`
+- HeapBuffer = `1`
+- Heap = `2`
```
Comment (vladbochok) [inline docs/api/js/types.md:239]: This is legacy that should be removed from both SDK and docs. I will prepare the PR for SDK, in the meanwhile please remove it

File: docs/api/js/types.md [hunk 5]
```diff
@@ -1,226 +1,303 @@
...
+- `logs`: `Array<Log>`;
+- `l2ToL1Logs`: `Array<L2ToL1Log>`;
+
+## `TransactionRequest`
+
+Transaction request type extended from the Ethers `providers.TransactionRequest` which adds an optional field for EIP-712 transactions.
+
+- `customData?`: `Eip712Meta`;
+
+## `TransactionResponse`
```
Comment (AnastasiiaVashchuk) [inline docs/api/js/types.md:284]: i would add info that this type extends providers.TransactionResponse from ethers

Comment (kailash-manasarovar) [reply docs/api/js/types.md:284]: DONE

File: docs/api/js/types.md (hunk 4 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md:239]: Done

File: docs/api/js/types.md [hunk 6]
```diff
@@ -1,226 +1,303 @@
...
+## `GeneralPaymasterInput`
+
+Interface representation of general paymaster input containing a couple of fields, including a `General` type.
+
+- `type`: `'General'`;
+- `innerInput`: `BytesLike`;
+
+## `L2ToL1Log`
+
+Interface representation of a layer 2 to layer 1 transaction log containing various fields.
```
Comment (AnastasiiaVashchuk) [inline docs/api/js/types.md:169]: Layer 2 to Layer 1

Comment (kailash-manasarovar) [reply docs/api/js/types.md:169]: why is it called `L2ToL1Log` ?

Comment (kailash-manasarovar) [reply docs/api/js/types.md:169]: we don't capitalize these words

File: docs/api/js/types.md [hunk 7]
```diff
@@ -1,226 +1,303 @@
...
+
+Non-enumerated enum list of transaction statuses.
+
+- NotFound = `not-found`
+- Processing = `processing`
+- Committed = `committed`
+- Finalized = `finalized`
+
+:::tip More info
+Find the code definition of the types above on the [zkSync Era Github repo.](https://github.com/matter-labs/zksync-2-dev/blob/98ed4fef2d62c28a7563b7874beaeb59392de211/sdk/zksync-web3.js/src/types.ts)
```
Comment (vladbochok) [inline docs/api/js/types.md:302]: This is a private repo, it is not accessible for everyone. Instead, use https://github.com/matter-labs/zksync-era

File: docs/api/js/types.md [hunk 8]
```diff
@@ -1,226 +1,303 @@
...
+
+Pipe-delimited list of block labels that includes block number in denary and hex plus block statuses.
+
+- `number`
+- `string` // hex representation of block number
+- `committed`
+- `finalized`
+- `latest`
+- `earliest`
+- `pending`
```
Comment (AnastasiiaVashchuk) [inline docs/api/js/types.md:90]: they should be fin into quotes like 
`'latest'`

**Comment by kailash-manasarovar** [comment]
Gonna make all these new updates in a new branch ... merged too quick :D @AnastasiiaVashchuk @vladbochok

File: docs/api/js/types.md [hunk 9]
```diff
@@ -1,226 +1,303 @@
...
-    initiatorAddress: Address;
-    receivedAt: Date;
-    ethCommitTxHash?: string;
-    ethProveTxHash?: string;
-    ethExecuteTxHash?: string;
-}
-```
+# Types and interfaces
+
+## `AccountAbstractionVersion`
```
Comment (AnastasiiaVashchuk) [inline docs/api/js/types.md:3]: nit: i would add more detailed  explelation what is the difference.
cc @StanislavBreadless

File: docs/api/js/types.md [hunk 10]
```diff
@@ -1,226 +1,303 @@
...
+
+Enumerated list of account abstraction versions.
```
Comment (AnastasiiaVashchuk) [inline docs/api/js/types.md:5]: same here

File: docs/api/js/types.md [hunk 11]
```diff
@@ -1,226 +1,303 @@
...
+- `proveTxHash?`: `string`;
+- `provenAt?`: `Date`;
+- `executeTxHash?`: `string`;
+- `executedAt?`: `Date`;
+- `l1GasPrice`: `number`;
+- `l2FairGasPrice`: `number`;
+
+## `Block`
+
+Interface representation of a block that extends the Ethers `providers.Block` definition with additional fields.
```
Comment (AnastasiiaVashchuk) [inline docs/api/js/types.md:57]: nit: I would added a link to `providers.Block` docs here, and in similar places as well

File: docs/api/js/types.md (hunk 9 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md:3]: I'm being precise, we called the doc Types previously but as I went through I realised they're not all types, they're interfaces and enums also .. I can't call an interface, or an enum a type

File: docs/api/js/types.md (hunk 7 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md:302]: DONE in new branch.

File: docs/api/js/types.md (hunk 8 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md:90]: DONE in new branch.

File: docs/api/js/types.md (hunk 9 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md:3]: Oh yes, there is one comment ... please check this in new branch.

File: docs/api/js/types.md (hunk 11 above)
Comment (kailash-manasarovar) [reply docs/api/js/types.md:57]: DONE for all

**Pull request merged**